src/features/code-push/hooks/useCodePush.ts

```diff
@@ -0,0 +1,107 @@
+import { useEffect, useState } from 'react';
+import { Alert } from 'react-native';
+import RNBootSplash from 'react-native-bootsplash';
+import CodePush, { CodePushOptions, Package } from 'react-native-code-push';
+
+type RemotePackageBinary = Package & {
+  updateAppVersion?: boolean;
+};
+
+const useCodePush = (options?: CodePushOptions) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [updateAvailable, setUpdateAvailable] = useState(false);
+  const [bundleVersion, setBundleVersion] = useState<string | null>(null);
+
+  const getCurrentBundleVersion = async () => {
+    try {
+      const currentBundle = await CodePush.getUpdateMetadata(CodePush.UpdateState.RUNNING);
+      setBundleVersion(`${currentBundle?.appVersion}-${currentBundle?.label}`);
+    } catch (error) {
+      console.log('error getting bundle info');
+    }
+  };
+
+  const checkPendingUpdate = async () => {
+    try {
+      const localPackage = await CodePush.getUpdateMetadata(CodePush.UpdateState.PENDING);
+      if (localPackage) {
+        // TODO: Restart the app or Prompt the user to restart, etc
+        console.log('An update is pending, consider restarting the app');
+      }
+    } catch (error) {
+      // TODO: Handle error on the UI
+      console.log('error check pending update');
+    }
+  };
+
+  const checkForUpdate = async () => {
+    try {
+      const remotePackage = await CodePush.checkForUpdate(
+        undefined,
+        ({ updateAppVersion, appVersion }: RemotePackageBinary) => {
+          if (updateAppVersion) {
+            return Alert.alert(
+              'Update available',
```

Comment:
As i18n has already been merged, WDYT about using localization for these strings??